lib/alchemy-pg_search.rb

```diff
@@ -39,14 +38,16 @@ def self.remove_page(page)
     #
     # @param page [Alchemy::Page]
     def self.index_page(page)
-      remove_page page
-
       page.update_pg_search_document
-      page.all_elements.includes(:ingredients).find_each do |element|
-        element.ingredients.select { |i| Alchemy::PgSearch.is_searchable?(i.type) }.each do |ingredient|
-          ingredient.update_pg_search_document
-        end
-      end
+
+      document = page.pg_search_document
+      return if document.nil?
+
+      ingredient_content = page.all_elements.includes(ingredients: {element: :page}).map do |element|
+        element.ingredients.select { |i| i.searchable? }.map(&:searchable_content).join(" ")
+      end.join(" ")
+
+      document.update(content: "#{document.content} #{ingredient_content}".squish)
```

Comment:
Maybe we need to `update_columns`

```suggestion
      document.update_columns(content: "#{document.content} #{ingredient_content}".squish)
```